x-pack/platform/plugins/shared/onechat/server/routes/agents.ts

```diff
@@ -6,96 +6,45 @@
  */
 
 import { schema } from '@kbn/config-schema';
-import { Subject, Observable } from 'rxjs';
-import { ServerSentEvent } from '@kbn/sse-utils';
-import { observableIntoEventSourceStream } from '@kbn/sse-utils-server';
-import { ChatAgentEvent } from '@kbn/onechat-common';
-import type { CallAgentResponse } from '../../common/http_api/agents';
+import { AgentType } from '@kbn/onechat-common';
+import type { ListAgentsResponse } from '../../common/http_api/agents';
 import { apiPrivileges } from '../../common/features';
+import { agentToDescriptor } from '../services/agents/utils';
 import type { RouteDependencies } from './types';
 import { getHandlerWrapper } from './wrap_handler';
 
 export function registerAgentRoutes({ router, getInternalServices, logger }: RouteDependencies) {
   const wrapHandler = getHandlerWrapper({ logger });
 
-  router.post(
+  router.get(
     {
-      path: '/internal/onechat/agents/invoke',
-      security: {
-        authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
-      },
+      path: '/api/chat/agents',
```

Comment:
with us now moving this to a public API, is there anything we need to do for feature flagging here? This wont show up on OpenAPI spec right?